catroid/src/org/catrobat/catroid/web/ServerCalls.java

```diff
@@ -344,6 +344,12 @@ public boolean registerOrCheckToken(String username, String password, String use
 			userEmail = emailForUiTests;
 		}
 
+		if (userEmail == null) { //restricted users
+			userEmail = "restricted";
```

Comment:
You surely tested it, but is there a problem if the userEmail isn't a valid email address?
[when-i-find-a-hard-coded-string-during-code-review](http://thecodinglove.com/post/111849339882/when-i-find-a-hard-coded-string-during-code-review) :wink: